processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -98,17 +97,19 @@
 
   private static final Function<Object, String> STRING_TRANSFORMER = new Function<Object, String>()
   {
+    @Nullable
     @Override
-    public String apply(final Object o)
+    public String apply(@Nullable final Object o)
     {
-      return String.valueOf(o);
+      return o == null ? null : String.valueOf(o);
```

Comment:
can you explain in your PR description why this line is not enough to fix the problem. What is the purpose of the remaining changes?